net-tools/netwatch/src/udp.rs

```diff
@@ -51,97 +68,810 @@ impl UdpSocket {
         Self::bind_raw(addr)
     }
 
+    /// Is the socket broken and needs a rebind?
+    pub fn is_broken(&self) -> bool {
+        self.is_broken.load(std::sync::atomic::Ordering::SeqCst)
+    }
+
+    /// Marks this socket as needing a rebind
+    fn mark_broken(&self) {
+        self.is_broken
+            .store(true, std::sync::atomic::Ordering::SeqCst);
+    }
+
+    /// Rebind the underlying socket.
+    pub fn rebind(&self) -> Result<()> {
+        debug!("rebinding {}", self.addr);
+        // Remove old socket
+        let mut guard = self.socket.write().unwrap();
+        {
+            let Some(socket) = guard.take() else {
+                bail!("cannot rebind closed socket");
```

Comment:
should state the reason: socket already closed or no socket to rebind or something like that.